docs/partitioning.md

```diff
@@ -1,39 +1,122 @@
-variants in chromosome 1.
+
+Running this query will process 5.37 TB which is the entire table. Running the
+same query on a clustered table (based on 3 columns: `reference_name,
+start_position, end_position`) will process only 551.09 MB. In terms
+of actual price (as of today BigQuery costs $5 per TB), we are looking at
+reduction from $26.85 to $0.0026.
+
+For quick reference, you can cluster your table using a query like this:
+
+```
+#standardSql
+CREATE TABLE `clustered_table`
+PARTITION BY fake_date
+CLUSTER BY reference_name, start_position, end_position AS (
+  SELECT *, DATE('1980-01-01') fake_date
+  FROM `original_table`
+)
+```
+
+Clustering can be very effective in reducing the query costs, however, it has a few limitations:
+ * Clustering does not offer any *guarantees* on the cost (it's a best effort reduction).
+ * It needs a one time clustering step which can take a few hours for large datasets.
+ * If you append data to an existing clustered table it will become partially sorted. you need to regularly re-clustering your table.
+
+## Solution2: Partitioning output table
+
+A second method for reducing the cost of BigQuery queries is to use
+Variant Transforms' native partitioning. Using this technique you can split the output table into
+several smaller tables, each containing variants of a specific region of the
+chromosome. For example, you can have one output table per chromosome, in that
+case the above query can be written as:
+
+```
+SELECT * from `1000genomes_chr17`
+WHERE start_position >= 41197694
+  AND end_position <= 41276113
+```
+
+Note that we remove the condition on the `reference_name` since we know this
+table, as its name suggests, only contains the variants of 17th chromosome.
+Unlike the BigQuery clustering, this method offers a cost guarantee.
+So by separating tables based on the `reference_name` of variants you are
+guaranteed that each chromosome's query will only process variants of that
+chromosome. Also, you can append new rows to your existing tables and still
+enjoy the same guarantee on the cost.
+
+Partitioning is not perfect either, you will be charged for processing of the full
+table when you query it, even though if you only process a subset of rows in
+the table. The previous query, for example, will cost 152 GB or $0.74. Even though this is
+significantly less than the cost without partitioning but it's more than
+clustering cost. That's why we offer the following solution as the most cost
+effective solution.
+
+## Solution3: Hybrid Solution
+
+This solution combines the previous two solutions together and it offers
+the benefits of both. Using the Variant Transforms' native partitioning
+output table will be split into several smaller tables (perhaps one table per
+chromosome) and then each table will be clustered based on the `start_position,
+end_position` columns to further optimize your queries. Using this technique
+the *BRC1 query* will cost 226 MB or $0.0011 which is more than half
```

Comment:
@samanvp Did you mean BRCA1?  BRC1 is a totally different gene in a completely different organism.